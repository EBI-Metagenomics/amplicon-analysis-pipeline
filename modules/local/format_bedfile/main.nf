
process FORMAT_BEDFILE {
    tag "$meta.id"
    label 'very_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
        "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"

    input:
    tuple val(meta), path(concat_ssu_lsu_coords)

    output:
    tuple val(meta), path("*maskfile.bed"), emit: format_bedfile_out

    script:
    """
    awk '\$2 > \$3 { var = \$3; \$3 = \$2; \$2 = var } 1 {print \$4,\$2,\$3}' OFS='\t' $concat_ssu_lsu_coords > ${meta.id}_maskfile.bed
    """

}
