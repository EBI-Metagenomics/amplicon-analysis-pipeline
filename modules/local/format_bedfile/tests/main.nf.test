nextflow_process {

    name "Test Process FORMAT_BEDFILE"
    script "../main.nf"
    process "FORMAT_BEDFILE"
    tag "modules"
    tag "format_bedfile"

    test("Should take ssu+lsu coordinates and return a formatted .bed file for masking them with bedtools") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('${projectDir}/modules/local/format_bedfile/tests/data/test_coords', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.format_bedfile_out).match() }
            )
            
        }

    }

}
