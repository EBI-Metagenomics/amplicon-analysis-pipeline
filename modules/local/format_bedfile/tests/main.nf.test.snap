{
    "Should take ssu+lsu coordinates and return a formatted .bed file for masking them with bedtools": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_maskfile.bed:md5,c4995d75a92f483fbecf983670ece17f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-19T21:59:10.713506"
    }
}